packages/gcf-utils/test/gcf-logger.ts

```diff
@@ -0,0 +1,64 @@
+// Copyright 2020 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import {GCFLogger, initLogger} from '../src/gcf-utils';
+import {describe, beforeEach, it} from 'mocha';
+import {ObjectWritableMock} from 'stream-mock';
```

Comment:
This seems like really good approach for testing. My one concern is if there is a possibility that there is some async work in the event loop, when our GCF function finishes, when we use the `sonic-boom` transport.

Is there any way we could get an `await`able Promise from `sonic-boom`, that resolves when logs have actually written to their intended destination? I think we should create a tracking issue, and test the heck out of the behavior of logs that occur immediately before a response from a cloud function.

Reply:
So there's 2 points here:

1. The way we're creating a logger right now is by passing the `{ sync: true }` option to `SonicBoom` which means that the logs will be flushed each time a log statement is written. So there's no async work happening in the background.

2. For logging within the `gcf()` function (coming in next PR) where we log the trigger info and github actions, we call `flushSync()` on SonicBoom at the end so that it synchronously flushes everything - so again, no async work.

This of course does come with some performance hit but in our case I think having all the logs is more important than performance (considering most bots are well below the GCF timeout). 

Either way, let's definitely test the heck out of this